src/controllers/legacycontrollermapping.h

```diff
@@ -144,6 +207,58 @@ class LegacyControllerMapping {
         return m_deviceDirection;
     }
 
+#ifdef MIXXX_USE_QML
+    /// Adds a custom QML module file to the list of controller modules for this mapping.
+    /// @param dirInfo A FileInfo of the directory or QML module
+    /// @param builtin If this is true, the script won't be written to the XML
+    virtual void addModule(const QFileInfo& dirInfo,
+            bool builtin = false) {
+        m_modules.append(QMLModuleInfo(
+                dirInfo,
+                builtin));
+        setDirty(true);
+    }
+
+    const QList<QMLModuleInfo>& getModules() const {
+        return m_modules;
+    }
+
+    /// @brief Adds a screen info where QML will be rendered.
+    /// @param identifier The screen identifier
+    /// @param size the size of the screen
+    /// @param targetFps the maximum FPS to render
+    /// @param msaa the MSAA value to use for render
+    /// @param splashoff the rendering grace time given when the screen is requested to shutdown
+    /// @param pixelFormat the pixel encoding format
+    /// @param endian the pixel endian format
+    /// @param reversedColor whether or not the RGB is swapped BGR
+    /// @param rawData whether or not the screen is allowed to reserve bare data, not transformed
+    virtual void addScreenInfo(const QString& identifier,
+            const QSize& size,
+            uint targetFps,
+            uint msaa,
+            std::chrono::milliseconds splashoff,
+            QImage::Format pixelFormat,
+            LegacyControllerMapping::ScreenInfo::ColorEndian endian,
+            bool reversedColor,
+            bool rawData) {
+        m_screens.append(ScreenInfo{identifier,
```

Comment:
Originally, I didn't want to have `ScreenInfo` too much spread because its attachment to a `legacy`. Please note that I looked into replacing the current methods with `addScreenInfo(ScreenInfo)` instead but the amount of work required to update the unit test is significant. If you really want me to make it `addScreenInfo(ScreenInfo)`, I will remove the unit tests

Reply:
I haven't looked at the test too much, but couldn't you fix it by mass-replacement?
eg use `addScreenInfo\(()\)` and replace with `addScreenInfo\(ScreenInfo \{$1\}\)` sorta?

Reply:
No because it looks like struct value wilcard for mock doesn't work the same, so `EXPECT_CALL(*mapping, addScreenInfo(_, _, _, _, _, _, _, _, _)).Times(0);` cannot be `EXPECT_CALL(*mapping, addScreenInfo(ScreenInfo{_, _, _, _, _, _, _, _, _}).Times(0);`

Reply:
(65 instances in the test)

Reply:
I'm sure it is doable, I was just questioning what is the value of changing arguments to struct, and not keep `ScreenInfo` as an internal representation used by `LegacyControlerMapping` to store data, since there is not performance gain from either approach AFAIK.

Now, since there seems to be a consensus on using that approach, I'm make the change as per requested

> I think that changing the interface to make a test easier is not a good practice

I disagree. I think that structuring your code in a way that it can easily be tested is good practice. I guess there is no wrong and write answer on that topic tho, so again, will change that as you've requested.

Reply:
Actually as I started the change, I realised another reason why I went this way, which is that `ScriptFileInfo` follows the same logic - a method with all required setting, and the building of a internal data representation using the struct. I guess you would want me to update that one too?

Reply:
Turns out there is a `FieldsAre` matcher for gmock, so I didn't spend as long as I feared.

I've updated both `Screeninfo` and `ScriptFileInfo` now so I hope everything is in order this time.